hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetTypeManagement.java

```diff
@@ -0,0 +1,229 @@
+
+    private final VirtualPropertyReplacer virtualPropertyReplacer;
+
+    private final Database database;
+    private final QuotaManagement quotaManagement;
+
+    /**
+     * Constructor
+     *
+     * @param targetTypeRepository
+     *            Target type repository
+     * @param targetRepository
+     *            Target repository
+     * @param virtualPropertyReplacer
+     *            replacer
+     * @param database
+     *            database
+     */
+    public JpaTargetTypeManagement(final TargetTypeRepository targetTypeRepository,
+            final TargetRepository targetRepository, final DistributionSetTypeRepository distributionSetTypeRepository,
+            final VirtualPropertyReplacer virtualPropertyReplacer, final Database database,
+            final QuotaManagement quotaManagement) {
+        this.targetTypeRepository = targetTypeRepository;
+        this.targetRepository = targetRepository;
+        this.distributionSetTypeRepository = distributionSetTypeRepository;
+        this.virtualPropertyReplacer = virtualPropertyReplacer;
+        this.database = database;
+        this.quotaManagement = quotaManagement;
+    }
+
+    @Override
+    public Optional<TargetType> getByName(String name) {
+        return targetTypeRepository.findByName(name).map(TargetType.class::cast);
+    }
+
+    @Override
+    public long count() {
+        return targetTypeRepository.count();
+    }
+
+    @Override
+    public TargetType create(TargetTypeCreate create) {
+        final JpaTargetTypeCreate typeCreate = (JpaTargetTypeCreate) create;
+        return targetTypeRepository.save(typeCreate.build());
+    }
+
+    @Override
+    public List<TargetType> create(Collection<TargetTypeCreate> creates) {
+        return creates.stream().map(this::create).collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    @Retryable(include = {
+            ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
+    public void delete(final Long targetTypeId) {
+        throwExceptionIfTargetTypeDoesNotExist(targetTypeId);
+
+        if (targetRepository.countByTargetTypeId(targetTypeId) > 0) {
+            throw new TargetTypeInUseException("Cannot delete target type that is in use");
+        }
+
+        targetTypeRepository.deleteById(targetTypeId);
+    }
+
+    @Override
+    public Slice<TargetType> findAll(Pageable pageable) {
+        return convertPage(targetTypeRepository.findAllWithoutCount(pageable), pageable);
+    }
+
+    @Override
+    public Page<TargetType> findByRsql(Pageable pageable, String rsqlParam) {
+        final Specification<JpaTargetType> spec = RSQLUtility.buildRsqlSpecification(rsqlParam, TargetTypeFields.class,
+                virtualPropertyReplacer, database);
+
+        return convertPage(targetTypeRepository.findAll(spec, pageable), pageable);
+    }
+
```

Comment:
Implement methods suggested in comment: https://github.com/eclipse/hawkbit/pull/1162/files#r690378578
Use specifications suggested in comment: https://github.com/eclipse/hawkbit/pull/1161/files#r690372459